efibootmgr: add page (#3788)
tldr-pages/tldr: pages/linux/efibootbgr.md

# efibootmgr

> Manipulate the UEFI Boot Manager (the Bootoptions).

> 

Reviewer: Also, let's add a "More Information" link to the man page.

- List the current settings:

> `efibootmgr -v for Filepaths`

Reviewer: Please use just the simple 'efibootmgr' as the first example. Make `-v` to be the last example.

> `efibootmgr `(`-v` for Filepaths)

Reviewer 2: The above command doesn't work for me, it needs another `, and even if I add it I the next traceback (syntax error) before the settings of my boot manager:

```
bash: command substitution: línea 2: error sintáctico: no se esperaba el final del fichero
bash: command substitution: línea 1: error sintáctico cerca del elemento inesperado `)'
bash: command substitution: línea 1: ` for Filepaths)'
```

The suggestion I do below works nice for me. It might be useful to have the feedback of another reviewers to test this.
```suggestion
`efibootmgr -v for Filepaths`
```

Reviewer 2: This one [has been reported](https://github.com/tldr-pages/tldr/pull/3788#issuecomment-577631625) tldr-bot.

> - Create new Bootoption:

Reviewer 2: ```suggestion
- Create new boot option:
```

"Boot option" should be in lowercase.

> `sudo efibootmgr -c -d {{esp_blockdevice}} -l {{\path\to\file.efi}} -L "{{Label}}"`

Reviewer: The convention for path tokens we follow is "path/to/file.efi". And please give a concrete example for block device - like "{{/dev/sda1}}" and same for label.

Author: The backslashes are there, because the ESP is a FAT Filesystem, and "officially" FAT uses \. `efibootmgr -v` is displaying the Paths using \, the Manpage is using \. And UEFI Shell also displays \.
We have had the same thing somewhere in this PR already.

Reviewer: The `/` convention is something we decided to use irrespective of what platform the page is for.

Reviewer 3: I think I remember pointing this one out before @agnivade before - in this case it's not a regular file path. It's an EFI file path, which _must_ use backward slashes - otherwise it doesn't work IIRC.

So in this case I think we might be able to make an exception.

> `sudo efibootmgr -c -d {{esp_blockdevice}} -l {{\path\to\file.efi}} -L "{{Label}}"`

Reviewer 3: I'm a bit confused - doesn't Linux use forward slashes `/` in paths - perhaps this will read better?
```suggestion
`sudo efibootmgr -c -d {{esp_blockdevice}} -l {{/path/to/file.efi}} -L "{{Label}}"`
```

Author: The ESP Blockdevice has a FAT (dos) Filesystem, and FAT Filesystems are ("officially") using \.
I have no clue if you could use / as well, because I have never done it. And `efibootmgr -v` also gives out file paths like  `Boot0001* GRUB	HD(1,GPT,[UUID],0x800,0x20000)/File(\EFI\GRUB\grubx64.efi)`.

Reviewer 3: Ah, I see. So this isn't a regular file path - it's a special ESP / boot filepath. That makes more sense then :-)

> - Add UEFI Shell v2 as a boot option:

Reviewer: This is the same as above. We can just make the concrete examples part of that itself. Let's remove this one.

Reviewer 4: Can you explain this more clearly? I don't quite understand you.

Reviewer 3: I think what @agnivade means here is that this example is identical to the previous one in functionality, but unlike the example that appears directly before this one - this example contains concrete values instead of using the `{{token_syntax}}`. Hence, we can instead edit the previous example to use concrete values inside the `{{token_syntax}}` - and remove this example.

Reviewer: That's exactly it.

Reviewer 3: Not sure what's happened here, but this example doesn't look identical to the one before it anymore to me. I'mconfused?

/cc @agnivade

Reviewer: This commit changed it https://github.com/tldr-pages/tldr/pull/3788/commits/f27372da2f82749602aedd8fed59cb563c80544a

> - > Specific Example: Adding UEFI Shell v2 as a Boot Options:

Reviewer 2: ```suggestion
- Add UEFI Shell v2 as a boot option:
```
It isn't necessary to specify "Specific example" nor ">". In addition, "Boot Options" should be in singular and in lowercase.

Reviewer 3: Marking as unresolved, since I can't see any changes to the file itself.

`sudo efibootmgr -c -d /dev/sda1 -l \EFI\tools\Shell.efi -L "UEFI Shell"`

> - Change the current boot order (you can get the Bootnums from the first command):

Reviewer: Let's mention in the first example that it shows the boot numbers. Then we don't need to mention that here.

`sudo efibootmgr -o {{0002,0008,0001,0005}}`

> - Delete Bootoption:

Reviewer 2: ```suggestion
- Delete boot option:
```

"Bootoption" should be in lowercase.

> `sudo efibootmgr -b {{bootnum}} -B`

Reviewer: Let's use the more descriptive "--delete-bootnum" instead of "-B" and use a concrete number like {{002}} for bootnum.

# efibootmgr

> Manipulate the UEFI Boot Manager (the Bootoptions).
> More information: https://linux.die.net/man/8/efibootmgr.

- List the current settings / bootnums:

`efibootmgr`

> - List the Filepaths:

Reviewer: Should Filepaths be capitalized ? I can make the change myself, but just wanted to check.

Author: As a German, I capitalize a lot of words, that the english language doesn't capitalize. 🙃

Thinking about it, I shouldn't be capitalized

Reviewer 2: I think the suggestion made by @agnivade is right and due it was a little change and @sp1ritCS seems to be agree, I performed the change.

# efibootmgr

> Manipulate the UEFI Boot Manager (the Bootoptions).

> >

Reviewer 3: Looks like this blank line is throwing off the linter. Perhaps we can remove it?
